sync/src/synchronizer/block_fetcher.rs

```diff
@@ -168,13 +168,17 @@ where
             let mut inflight = self.synchronizer.peers.blocks_inflight.write();
             let count = MAX_BLOCKS_IN_TRANSIT_PER_PEER
                 .saturating_sub(inflight.peer_inflight_count(&self.peer));
+            let max_height_header = self
```

Comment:
Why this extra step is required?

Reply:
A minor optimization. typical `get_ancestor` with 30W ancestors need about `60` steps, with this optimization will reduce to less than `(max_height - fixed_last_common_header.number())`